### PR TITLE
improve single precision support in skimage.exposure

### PR DESCRIPTION
**Pull request by grlee77**
## Description

I had missed this module when adding float32 support earlier!

A few of these functions already preserved floating point precision (`adjust_gamma`, `adjust_log`, `adjust_sigmoid`), so just needed test cases added. One question is if we should start forcing these functions to promote float16 to float32, for consistency with most other scikit-image functions.

A few others now just cast the output to float32 when float16 or float32 input is provided, but do not perform internal calculations in float32. For example, both `match_histograms` and `equalize_hist` use `np.interp` which always returns float64 arrays which I just cast back to single precision when desired. We could optionally leave these functions as they were as this extra cast adds a little overhead. I thought that overhead may be outweighed by the benefit of maintaining precision in a pipeline that includes these functions, though.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by rfezzani — APPROVED** [review]
Thank you @grlee77 :wink:

**Comment by grlee77** [comment]
Thanks for reviewing this @rfezzani. This just needs one more approval.

If we want to make `exposure.adjust_*` promote float16 to float32 we can address that in a separate PR later.

File: skimage/exposure/exposure.py [hunk 1]
```diff
@@ -428,7 +436,7 @@ def intensity_range(image, range_values='image', clip_negative=False):
     return i_min, i_max
 
 
-def _output_dtype(dtype_or_range):
+def _output_dtype(dtype_or_range, image_dtype):
```
Comment (mkcor) [inline skimage/exposure/exposure.py:439]: Shouldn't we add an `image_dtype` entry under `Parameters`?

File: skimage/exposure/tests/test_exposure.py [hunk 2]
```diff
@@ -255,10 +266,13 @@ def test_rescale_shrink():
     assert_array_almost_equal(out, [0, 0.5, 1])
 
 
-def test_rescale_in_range():
-    image = np.array([51., 102., 153.])
+@pytest.mark.parametrize("dtype", [np.float16, np.float32, np.float64])
```
Comment (mkcor) [inline skimage/exposure/tests/test_exposure.py:269]: There's a mix of `@pytest.mark.parametrize('dtype', [np.float16, np.float32, np.float64])` and `@pytest.mark.parametrize("dtype", [np.float16, np.float32, np.float64])` in these new/updated tests... :frowning:

Comment (grlee77) [reply skimage/exposure/tests/test_exposure.py:269]: I will update to consistently use single quotes for the new lines added in this PR. There is a mixture of single and double quotes elsewhere in the file, but probably not worth changing the unrelated stuff in this PR.

File: skimage/exposure/exposure.py (hunk 1 above)
Comment (grlee77) [reply skimage/exposure/exposure.py:439]: sure, I will add it

**Review by mkcor — APPROVED** [review]

**Pull request merged**